src/software/ai/hl/stp/action/action_test.cpp

```diff
@@ -69,3 +69,28 @@ TEST(ActionTest, restart_after_done_makes_done_false)
     EXPECT_TRUE(intent_ptr);
     EXPECT_FALSE(action.done());
 }
+
+TEST(ActionTest, action_with_loop_forever_does_not_return_nullptr)
+{
+    Robot robot = Robot(0, Point(), Vector(), Angle::zero(), AngularVelocity::zero(),
```

Comment:
super nit: I prefer not using the default constructors of `Point` and `Vector` where possible just to be super explicit that it's `(0, 0)`

Reply:
fair, ill make that change